pkg/async/worker_test.go

```diff
@@ -283,3 +283,78 @@ func TestWorkerReceiveAndWorkBlocksUntilContextCanceled(t *testing.T) {
 	err := w.receiveAndWork(ctx, getDisposableQueueName())
 	assert.Equal(t, ctx.Err(), err)
 }
+
+func TestWorkerDoesNotMoveTaskEarly(t *testing.T) {
```

Comment:
I know the names of some of our test functions are unpleasantly long, but I want to make sure they're sufficiently descriptive. A worker is comprised of multiple components, so can this be something like `TestDelayedQueueWatcherDoesNotMoveTaskEarly`? Similar though for the other new test.